### DISCUSSION TITLE
How to use dictionary logging when no backends are enabled?

### DISCUSSION THREAD

**Discussion by ashwinyes** [zephyrproject-rtos/zephyr#51161] [Q&A]
We have a use case where we require to log to a buffer. The buffer will be drained out by a host and Zephyr need not bother about this part.

I would like to enable CONFIG_LOG_FRONTEND and CONFIG_LOG_FRONTEND_ONLY, and implement my own versions of log_frontend_msg(), log_frontend_panic() and log_frontend_init() which will be logging to the buffer. So I don't require any backends. 

Here I am not able to enable dictionary logging as it has a dependency on one of the backends which supports it.

Why is dictionary logging not supported when only frontend is required?

**Comment by yashi** [comment]
The help message says:

> Dictionary based logging is binary based where predefined strings are
> replaced with their equivalent addresses from the built binary image
> file in log output. This reduces the size required to store the log
> output when there are long format strings to be logged.
> 
> This should be selected by the backend automatically.
